lib/driver.js

```diff
@@ -382,7 +380,7 @@ class XCUITestDriver extends BaseDriver {
   }
 
   async startWda (sessionId, realDevice) {
-    this.wda = new WebDriverAgent(this.xcodeVersion, this.opts);
+    this.wda = new WebDriverAgent(this.xcodeVersion, Object.assign(this.opts, {isFirstSession: this.isFirstSession}));
```

Comment:
I think this should be ok. Object.assign guarantees, that WebDriverAgent will not modify any properties, which is true, since we anyway reassign the necessary properties in opts to WebDriverAgent's own property values.